### DISCUSSION TITLE
Is it me or is Cavernize having low accuracy?

### DISCUSSION THREAD

**Discussion by EaryChow** [VoidXH/Cavern#194] [General]
I have been trying out Cavernize and I cannot get it to work as I hope it would. I have thought about reporting this as a bug but I am not sure whether this is a bug or this is just Cavernize's inherent limitation. 

Attached is a zip file containing a 9.1.6 channel check, including the master file, an MKV containing ec3, and an MKV containing an eb3. (BTW do you know whether eb3 is supposed to contain 7.1 or 5.1.2 downmix?)

[916_ENG.zip](https://github.com/user-attachments/files/17675438/916_ENG.zip)

I am trying to run the MKVs through Cavernize for it to spit out a 9.1.6 multichannel file, but it seems weird.

In Cavernize GUI I chose the output format to be "9.1.6 with top sides", and I import the result into my DAW to take a look.

In eb3's case, it seems Cavernize completely silenced the surround and the rear surround channels, as well as the top middle and the top rear channels, just silence. 

In ec3's case, the floor channels are more or less accurate, the only thing is the wide channels somehow leaking into surround channels, but I suspect that might be Dolby's bed and object encoding doing something weird, and it's overall fine. 

But when it comes to sky channels, it's a mess. the left top front audio comes through to four sky speakers all at once, the rest of the sky channels are more or less like that, they are seriously leaked.

Is it me? Or is it Cavernize not being accurate enough? Is it an inherent limitation that cannot be improved, or is it something that will improve in the future?

**Comment by VoidXH** [comment]
There are multiple issues with the source content:

First, single audio source content can be encoded with sparse JOC, which is not properly documented by Dolby and can't be decoded. Cavernize shows you an error regarding this after export.

Second, the objects are at wrong positions. The top objects are in the middle (see with Cavern Driver's player), they are correctly mapped to multiple top channels. While Dolby has a 2x scale to all object positions to exaggerate the spatialization, Cavern aims for precision, and this resulted in many times a noticeably improved audio quality, like for #91.

**Comment by EaryChow** [comment]
> Cavernize shows you an error regarding this after export.

I do see it sometimes but not this file, I don't see the warning when converting this file. So no this is not the case, your comment doesn't apply to this file.

> Second, the objects are at wrong positions

Doesn't make sense, I put each channel as their own tracks, and link all channels as 9.1.6, the DAW automatically assigns the correct panning, and I send them to the renderer. Are you suggesting my DAW, Fairlight in Davinci Resolve, uses the wrong speaker layout position? It's kind of hard to believe.

Assuming Blackmagic is doing the right thing, it's very likely Cavern/Cavernize is the one being wrong. If it's the other way around, can I see some more info on that? Maybe some Dolby Documentation? In that case I will have to go to Blackmagic forum to ask their reason for chosing the wrong speaker layout position.

BTW this is 9.1.6 speaker layout in Resolve Fairlight:
![image](https://github.com/user-attachments/assets/f6fee719-7e3b-4cdb-a5b6-cb597332205d)

**Reply by VoidXH** [reply]
> I do see it sometimes but not this file, I don't see the warning when converting this file. So no this is not the case, your comment doesn't apply to this file.

Correct, root cause found, bug #196 was created for this.

> BTW this is 9.1.6 speaker layout in Resolve Fairlight

Which is incorrect. Standard Dolby mixes have the objects at the corners of the room. Fairlight most likely choose these locations as they get mapped to the sides on Dolby renderers. However, mixes still use the cinema standard corners, and for smoothness, they are preferred.

If you need that kind of rendering, you can reorder the channels in the Cavern Driver to these locations and export with Cavernize using the Cavern Driver layout.

**Reply by EaryChow** [reply]
I posted the issue on Blackmagic forum: https://forum.blackmagicdesign.com/viewtopic.php?f=21&t=212549#p1102122 
They responded that Fairlight's speaker placement is per Dolby's recommendation. I am perplexed, what exactly is the standard as proposed by Dolby? All I can find is some documents writing the angle degree range for each speakers (like 7.1.4's Ls speaker can be placed between 90 to 110 degrees), not a word about whether the sky speakers should be in the corner or somewhere closer to the middle.

**Reply by VoidXH** [reply]
Looks like they worked with those angle documents, which is the placement guide, but not the theatrical targets. These are described by multiple vendors, including Dolby, in multiple standards, at the corners. Even Dolby E-AC-3 in ETSI TS 103 420 Table B.10 describes the corners. Everyone follows it, with the exception of Fairlight.
